homeassistant/components/media_player/firetv.py

```diff
@@ -0,0 +1,214 @@
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the firetv platform. """
+    add_devices([
+        FireTVDevice(
+            config.get('host', 'localhost:5556'),
+            config.get('device', 'default'),
+            config.get('name', 'Amazon Fire TV')
+        )
+    ])
+
+
+class FireTV(object):
+    """ firetv-server client.
+
+    Should a native Python 3 ADB module become available,
+    python-firetv can support Python 3, it can be added
+    as a dependency, and this class can be dispensed of.
+
+    For now, it acts as a client to the firetv-server
+    HTTP server (which must be running via Python 2).
+    """
+
+    DEVICE_STATE_URL = 'http://{0}/devices/state/{1}'
+    DEVICE_ACTION_URL = 'http://{0}/devices/action/{1}/{2}'
+
+    def __init__(self, host, device_id):
+        self.host = host
+        self.device_id = device_id
+
+    @property
+    def state(self):
+        """ Get the device state.
+
+        An exception means UNKNOWN state.
+        """
+        try:
+            response = requests.get(
+                FireTV.DEVICE_STATE_URL.format(
+                    self.host,
+                    self.device_id
+                    )
+                )
+            return response.json()['state']
+        except requests.exceptions.HTTPError:
+            return STATE_UNKNOWN
+        except requests.exceptions.RequestException:
+            return STATE_UNKNOWN
+
+    def action(self, action_id):
+        """ Perform an action on the device.
+
+        There is no action acknowledgment, so exceptions
+        result in a pass.
+        """
+        try:
+            requests.get(
+                FireTV.DEVICE_ACTION_URL.format(
+                    self.host,
+                    self.device_id,
+                    action_id
+                    )
+                )
+        except requests.exceptions.HTTPError:
+            pass
+        except requests.exceptions.RequestException:
+            pass
+
+
+class FireTVDevice(MediaPlayerDevice):
+    """ Represents an Amazon Fire TV device on the network. """
+
+    def __init__(self, host, device, name):
+        self._firetv = FireTV(host, device)
+        self._name = name
+
+    @property
+    def name(self):
+        """ Get the device name. """
+        return self._name
+
+    @property
+    def should_poll(self):
+        """ Device should be polled. """
+        return True
+
+    @property
+    def supported_media_commands(self):
+        """ Flags of media commands that are supported. """
+        return SUPPORT_FIRETV
+
+    @property
+    def state(self):
+        """ State of the player. """
+        state_map = {
+            'idle': STATE_IDLE,
+            'off': STATE_OFF,
+            'play': STATE_PLAYING,
+            'pause': STATE_PAUSED,
+            'standby': STATE_STANDBY,
+            'disconnected': STATE_UNKNOWN,
+        }
+        return state_map.get(self._firetv.state, STATE_UNKNOWN)
```

Comment:
State can be called multiple times and should not result in a query to the device. Put this logic in the `update` method instead, which will be automatically called by HA.